flow/layers/transform_layer_unittests.cc

```diff
@@ -224,5 +225,114 @@ TEST_F(TransformLayerTest, NestedSeparated) {
                    MockCanvas::DrawCall{1, MockCanvas::RestoreData{0}}}));
 }
 
+#ifdef FLUTTER_ENABLE_DIFF_CONTEXT
+
+using TransformLayerLayerDiffTest = DiffContextTest;
+
+TEST_F(TransformLayerLayerDiffTest, Transform) {
+  auto path1 = SkPath().addRect(SkRect::MakeLTRB(0, 0, 50, 50));
+  auto m1 = std::make_shared<MockLayer>(path1);
+
+  auto transform1 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeTrans(10, 10));
+  transform1->Add(m1);
+
+  LayerTree t1;
+  t1.root()->Add(transform1);
+
+  auto damage = DiffLayerTree(t1, LayerTree());
+  EXPECT_EQ(damage.surface_damage, SkIRect::MakeLTRB(10, 10, 60, 60));
+
+  auto transform2 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeTrans(20, 20));
+  transform2->Add(m1);
+  transform2->AssignOldLayer(transform1.get());
+
+  LayerTree t2;
+  t2.root()->Add(transform2);
+
+  damage = DiffLayerTree(t2, t1);
+  EXPECT_EQ(damage.surface_damage, SkIRect::MakeLTRB(10, 10, 70, 70));
+
+  auto transform3 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeTrans(20, 20));
+  transform3->Add(m1);
+  transform3->AssignOldLayer(transform2.get());
+
+  LayerTree t3;
+  t3.root()->Add(transform3);
+
+  damage = DiffLayerTree(t3, t2);
+  EXPECT_EQ(damage.surface_damage, SkIRect::MakeEmpty());
+}
+
+TEST_F(TransformLayerLayerDiffTest, TransformNested) {
+  auto path1 = SkPath().addRect(SkRect::MakeLTRB(0, 0, 50, 50));
+  auto m1 = CreateContainerLayer(std::make_shared<MockLayer>(path1));
+
+  auto transform1 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeScale(2.0, 2.0));
+
+  auto transform1_1 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeTrans(10, 10));
+  transform1_1->Add(m1);
+  transform1->Add(transform1_1);
+
+  auto transform1_2 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeTrans(100, 100));
+  transform1_2->Add(m1);
+  transform1->Add(transform1_2);
+
+  auto transform1_3 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeTrans(200, 200));
+  transform1_3->Add(m1);
+  transform1->Add(transform1_3);
+
+  LayerTree l1;
+  l1.root()->Add(transform1);
+
+  auto damage = DiffLayerTree(l1, LayerTree());
+  EXPECT_EQ(damage.surface_damage, SkIRect::MakeLTRB(20, 20, 500, 500));
+
+  auto transform2 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeScale(2.0, 2.0));
+
+  auto transform2_1 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeTrans(10, 10));
+  transform2_1->Add(m1);
+  transform2_1->AssignOldLayer(transform1_1.get());
+  transform2->Add(transform2_1);
+
+  auto transform2_2 =
+      std::make_shared<TransformLayer>(SkMatrix::MakeTrans(100, 101));
```

Comment:
Is 101 a typo?

Reply:
No, it is there on purpose. I needs to be slightly different than transform1_2 otherwise they are same and the diff is empty. I'll comment it.